City-of-Helsinki/drupal-helfi-platform-config
# UHF-6491: eu_cookie_compliance update

# [UHF-6491](https://helsinkisolutionoffice.atlassian.net/browse/UHF-6491)
Updated `eu_cookie_compliance` module.

## What was done
* Updated the minimum version for `eu_cookie_compliance` module to the latest stable release 1.24.
* Removed redundant old patch.
* Updated the other old patch:
  * It now applies to the updated module.
  * It was updated to use core/once instead of jQuery.once. The updated JS file is added by the patch and it uses the JS file provided by the module as a dependency. As the dependency JS file was updated to use core/once, the same change was made to the patch JS file.
  * When updated the module from 1.19 to 1.24, the module began to use a minified version of the main JS file. However, the patch file makes a change to the un-minified version of the file. Thus, the patch was updated so that the module still uses the un-minified version. Without this, the cookie popup functionality would silently break. (The sites are still minifying all JS files.)
* Added missing JS dependencies to the helfi_toc module.
  * As the jQuery.once was no longer included by the eu_cookie_compliance module nor by the patch, the unrelated toc module would break because the needed JS files were not included to the page.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-6491-eu-cookie-compliance-update -W`
* Run `make drush-updb drush-cr`

## How to test
* Assuming you have existing cookie selection for the site, test that the site works with that existing cookie data.
  * Check the selected cookie categories using the browser's dev tools.
  * When browsing the site, you shouldn't see the cookie popup.
  * The cookie settings page `/cookie-information-and-settings` shows the correct cookie selection.
  * When visiting a page that contains an embedded video, it still respects the cookie settings.
  * You can still change the cookie categories using the settings page.

* Clear the cookie settings using dev tools and reload the page.
  * You should now have the cookie popup visible. Both buttons should still work and the changes are visible at the cookie settings page `/cookie-information-and-settings`.
  * You can still make changes to cookie settings using the cookie settings page.
  * When visiting a page that contains an embedded video, it still respects the cookie settings.

* Clear the cookies once more and select essential cookies only.
  * Go to a page that contains a chat. Opening the chat should opt-in to chat cookies. Check this from the cookies settings page.

* Check that the unrelated TOC module still works by opening/creating a content page that includes table of contents. Check that there are no JS errors in the console.
